### PR TITLE
Feature/cleanup

### PR DESCRIPTION
**Pull request by morfj**
Refactoring and cleanup to make the interface of the classes more explicit and to improve test coverage a bit.

### CODE REVIEW

File: lib/wallet.js [hunk 1]
```diff
@@ -68,75 +77,79 @@ class Wallet extends ethers.Signer {
...
      * @param [options]
-     * @returns {Promise} A promise that resolves into a transaction with a hash.
+     * @return {Promise} A promise that resolves into a transaction with a hash.
      * @example
      * const {hash} = await wallet.completeTokenDepsoit('1.1', 'HBT', {gasLimit: 200000});
      * const receipt = await wallet.provider.getTransactionConfirmation(hash);
      */
     async completeTokenDeposit(amount, symbol, options) {
-        options = { gasLimit: 600000, ...options };
+        options = {gasLimit: 600000, ...options};
```
Comment (jonbern) [inline lib/wallet.js]: This is repeated a few times. Would it make sense to have it defined as a constant somewhere?

File: Docs/wallet.md [hunk 2]
```diff
@@ -20,140 +29,20 @@ A class for performing various operations on a wallet.
...
 | provider | <code>NahmiiProvider</code> | A NahmiiProvider instance |
 
-**Wallet from private key example**
-```js
-const privateKey = '0x9616c2ab6330c7fda535042c120b55d992fa8c2c2a3d82603ea043aeb09ff411';
-const provider = new NahmiiProvider(...);
-const softwareWallet = new Wallet(privateKey, provider);
-```
-
-**Wallet from Ledger Nano S example**
```
Comment (liamaharon) [inline Docs/wallet.md]: The hardware wallet examples are removed. Should we create maybe a new .md file with these examples?

Comment (morfj) [reply Docs/wallet.md]: Docs/*.md are generated files from JSDOC comments. Generation should be added to the npm "build" script.

Comment (morfj) [reply Docs/wallet.md]: For large code examples would it makes sense to have a separate folder called "examples"?

**Pull request merged**